modules/apps/dispatch/dispatch-talend-web/src/main/java/com/liferay/dispatch/talend/web/internal/display/context/DispatchTalendDisplayContext.java

```diff
@@ -1,4 +1,41 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
 package com.liferay.dispatch.talend.web.internal.display.context;
 
+import com.liferay.dispatch.metadata.DispatchTriggerMetadata;
+import com.liferay.dispatch.metadata.DispatchTriggerMetadataProvider;
+
+/**
+ * @author MahmoudAzzam
+ * @author Mahmoud Azzam
+ */
 public class DispatchTalendDisplayContext {
-}
+
+	public DispatchTalendDisplayContext(
+		DispatchTriggerMetadataProvider dispatchTriggerMetadataProvider) {
+
+		_dispatchTriggerMetadataProvider = dispatchTriggerMetadataProvider;
+	}
+
+	public DispatchTriggerMetadata getDispatchTriggerMetadata(
+		long dispatchTriggerId) {
+
+		return _dispatchTriggerMetadataProvider.getDispatchTriggerMetadata(
+			dispatchTriggerId);
+	}
+
+	private final DispatchTriggerMetadataProvider
+		_dispatchTriggerMetadataProvider;
+
```

Comment:
@MahmoudAzzam if you do not close Java source file with closing curly bracket compiler generates compile time error

Reply:
pay attention to it